src/regex.nim

```diff
@@ -2100,6 +2101,51 @@ proc groupsCount*(m: RegexMatch): int =
   ##
   m.groups.len
 
+
+proc groupsNames*(m: RegexMatch): seq[string] =
+  ## return the names of capturing groups
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupsCount == 2
```

Comment:
Rename it to `groupNames` (remove the S in groups) and use it in the example. I think the description should say `Return the name of the capturing groups`. 

Also, the example does not compile, I don't think any of the examples do.